src/main/kotlin/com/wgslfuzz/semanticspreservingtransformations/Common.kt

```diff
@@ -35,12 +35,14 @@ import kotlin.math.max
 private const val LARGEST_INTEGER_IN_PRECISE_FLOAT_RANGE: Int = 16777216
 
 interface FuzzerSettings {
-    val maxDepth: Int
-        get() = 5
+    fun goDeeper(currentDepth: Int): Boolean = randomDouble() < 1.0 / (currentDepth.toDouble() + 1.0)
```

Comment:
This looks good, but:

- The probability of initially going deeper is almost 100% (you'd have to pick 1.0 to not go deeper, for which the probability is really small)

- The probability of going deeper than 1 is only 0.5, which is perhaps a bit low.

Could you take an action to play with this distribution a bit in practice, having a look at the kinds of programs that come out?